acceptance/lib/puppet/acceptance/i18n_utils.rb

```diff
@@ -6,8 +6,9 @@ module I18nUtils
       #
       # @param agent [string] the agent to check the locale configuration on
       # @param language [string] the language attempt to configure if needed
+      # @param prefer_utf8 [Boolean] whether to prefer the utf-8 version of the locale when available
```

Comment:
I'm not sure I understand -- `locale -a` lists all available locales on stdout. This method looks at that output to try to find the requested locale. If the requested locale isn't in the output, it tries to generate the locale using `locale-gen`. Then the `language_name` method then runs `locale -a` again to try to find it (and prioritize UTF-8 for some (other) tests that require that behavior). I guess it's unlikely that any systems we test on will _not_ have the en_US locale, but I used this helper because it was already available.

Reply:
Yes, my question is how the `prefer_utf8` option affects the existing locale-enabling behavior. The diff shows that it is only used for stdout matching in `language_name` to find the preferred language string. It returns nil if that language string doesn't exist. I would find it strange for it to indicate that `{language}.utf8` or `{language}.UTF-8` exists but `{language}` doesn't (which seems to be the one use-case I can think of where setting the option to false would be useful.

My point is that I don't think this option is necessary here. Were the tests failing without it?

Reply:
Yes -- the bug here occurs when files are saved as UTF-8 but the current locale does not support UTF-8. That means that to duplicate the problem, we need to use a locale that does not support UTF-8, like en_US.

Puppet's CI hosts are mostly set up to use UTF-8 locales by default, so I looked for an existing method to set the locale to something else temporarily during this test. This method seemed appropriate, but it would always prioritize the UTF-8 locale, which didn't help duplicate the problem (the tests would always pass). 

So yes, the tests were passing on 5.5.x without this, but only because they were using a UTF-8 locale (and not duplicating the bug). Setting a non-UTF-8 locale makes 5.5.x fail these tests.

Reply:
Ah cool, thanks for clarifying why the method's used. I still don't see why the `prefer_utf8` option is necessary -- you can enable a non-UTF8 locale without it.

Reply:
Can you say more about that? The method always prioritized UTF-8 before -- are you suggesting I stop using the method and run `locale -a`/`locale-gen` in the test directly instead?

Reply:
Sure, I added a comment about that. The TLDR is that using the method is fine, I just don't see the need to extend it with the `prefer_utf8` option since it does not modify the existing behavior (and I outline why in that comment).